app/controllers/application_controller.rb

```diff
@@ -31,7 +31,18 @@ def analytics
 
   def after_sign_in_path_for(resource)
     analytics.track_event('Authentication Successful')
-    stored_location_for(resource) || session[:saml_request_url] || profile_index_path
+    safe_stored_location_for(resource) || profile_index_path
+  end
+
+  def safe_stored_location_for(resource)
+    url = stored_location_for(resource) || session[:saml_request_url]
+    if url =~ /\?SAMLRequest/
+      uri_params = Rack::Utils.parse_nested_query(URI.parse(url).query).symbolize_keys
+      saml_request = SamlIdp::Request.from_deflated_request(uri_params[:SAMLRequest])
+      url if saml_request.valid?
```

Comment:
I don't understand what the point of this is.  Won't the target url handle checking the validity of the saml_request?  Why do we need to handle it here?